src/client/Microsoft.Identity.Client/AppConfig/ApplicationConfiguration.cs

```diff
@@ -131,6 +131,25 @@ public string ClientVersion
         internal IRetryPolicyFactory RetryPolicyFactory { get; set; }
         internal ICsrFactory CsrFactory { get; set; }
 
+        #region Extensibility Callbacks
+
+        /// <summary>
+        /// Dynamic certificate provider callback for client credential flows.
+        /// </summary>
+        public Func<IAppConfig, X509Certificate2> ClientCredentialCertificateProvider { get; set; }

```

Comment:
Let's make all our extensibility points async from the start.

Reply:
Done